src/saltext/vmware/modules/esxi.py

```diff
@@ -1092,6 +1093,521 @@ def remove_user(
+    network_type="static",
+    datacenter_name=None,
+    cluster_name=None,
+    host_name=None,
+    service_instance=None,
+):
+    """
+    Create VMKernel Adapter on matching ESXi hosts.
+
+    port_group_name
+        The name of the port group for the VMKernel interface. (required).
+
+    dvswitch_name
+        The name of the vSphere Distributed Switch (vDS) where to add the VMKernel interface.
+        One of dvswitch_name or vswitch_name is required.
+
+    vswitch_name
+        The name of the vSwitch where to add the VMKernel interface.
+        One of dvswitch_name or vswitch_name is required.
+
+    enable_fault_tolerance
+        Enable Fault Tolerance traffic on the VMKernel adapter. Valid values: "True", "False".
+
+    enable_management_traffic
+        Enable Management traffic on the VMKernel adapter. Valid values: "True", "False".
+
+    enable_provisioning
+        Enable Provisioning traffic on the VMKernel adapter. Valid values: "True", "False".
+
+    enable_replication
+        Enable vSphere Replication traffic on the VMKernel adapter. Valid values: "True", "False".
+
+    enable_replication_nfc
+        Enable vSphere Replication NFC traffic on the VMKernel adapter. Valid values: "True", "False".
+
+    enable_vmotion
+        Enable vMotion traffic on the VMKernel adapter. Valid values: "True", "False".
+
+    enable_vsan
+        Enable VSAN traffic on the VMKernel adapter. Valid values: "True", "False".
+
+    mtu
+        The MTU for the VMKernel interface.
+
+    network_default_gateway
+        Default gateway (Override default gateway for this adapter).
+
+    network_type
+        Type of IP assignment. Valid values: "static", "dhcp".
+
+    network_ip_address
+        Static IP address. Required if type = 'static'.
+
+    network_subnet_mask
+        Static netmask required. Required if type = 'static'.
+
+    network_tcpip_stack
+        The TCP/IP stack for the VMKernel interface. Valid values: "default", "provisioning", "vmotion", "vxlan".
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.create_vmkernel_adapter port_group_name=portgroup1 dvswitch_name=dvs1
+    """
+    log.debug("Running vmware_esxi.create_vmkernel_adapter dvswitch_name=dvs1")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
```

Comment:
get_hosts is an existing salt util function with this signature. It can either filter by a list of hosts or take a flag to return all hosts in a datacenter/cluster. No explicit relationship here.

Reply:
Sounds good.